common/utils/src/main/resources/error/error-classes.json

```
@@ -133,6 +133,48 @@
       "Could not load Protobuf class with name <protobufClassName>. <explanation>."
     ]
   },
+  "CANNOT_LOAD_STATE_STORE" : {
+    "message" : [
+      "An error occurred during loading state."
+    ],
+    "subClass" : {
+      "CANNOT_READ_CHECKPOINT" : {
+        "message" : [
+          "Cannot read RocksDB checkpoint metadata of version <versionLine>"
+        ]
+      },
+      "CANNOT_READ_DELTA_FILE" : {
+        "message" : [
+          "Error reading delta file <fileToRead> of <clazz>: <message>"
+        ]
+      },
+      "CANNOT_READ_SNAPSHOT_FILE" : {
+        "message" : [
+          "Error reading snapshot file <fileToRead> of <clazz>: <message>"
+        ]
+      },
+      "CANNOT_READ_STREAMING_STATE_FILE" : {
+        "message" : [
+          "Error reading streaming state file of <fileToRead> does not exist. If the stream job is restarted with a new or updated state operation, please create a new checkpoint location or clear the existing checkpoint location."
+        ]
+      },
+      "UNEXPECTED_FILE_SIZE" : {
+        "message" : [
+          "Copied <dfsFile> to <localFile>, expected <expectedSize> bytes, found <localFileSize> bytes."
+        ]
+      },
+      "UNRELEASED_THREAD_ERROR": {
+        "message" : [
+          "<loggingId>: RocksDB instance could not be acquired by <newAcquiredThreadInfo> as it was not released by <acquiredThreadInfo> after <timeWaitedMs> ms.",
+          "Thread holding the lock has trace: <stackTraceOutput>"
+        ]
+      },
+      "WRAPPER" : {
+        "message" : [""]
+      }
+    },
+    "sqlState" : "40000"
```

```suggestion
    "sqlState" : "58030"
```

It's used by PostgreSQL and Redshift for io errors and classified as a "system error" but an internal error....